src/mockServiceWorker.js

```diff
@@ -79,6 +80,11 @@ self.addEventListener('message', async function (event) {
         self.registration.unregister()
       }
 
+      // Clear pending requests
+      // Otherwise this causes previous requests to resolve after a refresh
+      pendingRequests.forEach(clearTimeout)
+      pendingRequests.clear()
```

Comment:
I'm curious to see how the worker/clients will behave with this. Technically, even if we clear the response resolving timeouts, the response promise that we pass to `event.respondWith` till still be pending. I may be wrong on this, though. 

Giving a little heads up: I'm reworking our integration tests to use `page-with`. Planning to merge that today/tomorrow. That should make the integration test writing a slightly better experience.

Reply:
While I'm trying to write tests for this, I'm thinking of a different solution that might address your valid concerns because that promise will still be pending.
Instead of clearing the timeout, we could also invoke the `reject` method of the promise.

In the test itself this does give the error `FetchError: request to https://test.api/ failed, reason: getaddrinfo ENOTFOUND test.api` tho :( . I'll also try this approach in the reproduction repo's.

Reply:
Seems like rejecting the promise won't work, this gives the following result in the vue repo (https://github.com/sumerokr/vue-mswjs)

```
(node:11268) UnhandledPromiseRejectionWarning: FetchError: invalid json response body at https://example.com/user reason: Unexpected token < in JSON at position 0
(node:11268) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 3)
```

Reply:
Perhaps we should `resolve` it instead? We'd have to provide it with a valid `Response` instance, otherwise the `event.respondWith` will raise an exception. Providing it with an empty response should suffice:

```js
// When clearing pending response promises:
resolve(new Response())
```